src/frontend/planner_test/tests/testdata/watermark.yaml

```diff
@@ -77,6 +77,25 @@
       | └─StreamTableScan { table: t1, columns: [t1.ts, t1.v1, t1.v2, t1._row_id], pk: [t1._row_id], dist: UpstreamHashShard(t1._row_id) }
       └─StreamExchange { dist: HashShard(t2.ts) }
         └─StreamTableScan { table: t2, columns: [t2.ts, t2._row_id], pk: [t2._row_id], dist: UpstreamHashShard(t2._row_id) }
+- name: band hash join
+  sql: |
+    create table t1 (ts timestamp with time zone, v1 int, v2 int, watermark for ts as ts - INTERVAL '1' SECOND) append only;
```

Comment:
Some suggestion: Maybe it would also be helpful to add a test that evaluates the query and not just looks at the plan. Inspired by the PSQL tests: 

```sql
CREATE TABLE t1 (
	c1 int,
	c2 int,
	c3 int,
	c4 timestamptz,
	c5 timestamp,
	c6 varchar,
	c7 varchar,
	c8 int,
	CONSTRAINT t1_pkey PRIMARY KEY (c1)
);
CREATE TABLE t2 (
	c1 int,
	c2 text,
	CONSTRAINT t2_pkey PRIMARY KEY (c1)
);

CREATE TABLE t4 (
	c1 int,
	c2 int,
	c3 text,
	CONSTRAINT t4_pkey PRIMARY KEY (c1)
);

INSERT INTO t4
	SELECT id,
	       id + 1,
	       'AAA'
	FROM generate_series(1, 100, 1) as t(id);
DELETE FROM t4 WHERE c1 % 3 != 0;

INSERT INTO t1
	SELECT id,
	       id % 10,
	       id,
	       '1970-01-01'::timestamptz + ((id % 100) || ' days')::interval,
	       '1970-01-01'::timestamp + ((id % 100) || ' days')::interval,
	       id % 10,
	       id % 10,
	       1
	FROM generate_series(1, 1000, 1) as t(id);

INSERT INTO t2
	SELECT id,
	       'AAA'
	FROM generate_series(1, 100, 1) t(id);

SELECT t1.c1, t2.c1 FROM (SELECT c1 FROM t4 WHERE c1 between 50 and 60) t1 FULL JOIN t2 ON (t1.c1 = t2.c1) ORDER BY t1.c1, t2.c1;
```

Reply:
> Some suggestion: Maybe it would also be helpful to add a test that evaluates the query and not just looks at the plan. Inspired by the PSQL tests:
> 
> ```sql
> CREATE TABLE t1 (
> 	c1 int,
> 	c2 int,
> 	c3 int,
> 	c4 timestamptz,
> 	c5 timestamp,
> 	c6 varchar,
> 	c7 varchar,
> 	c8 int,
> 	CONSTRAINT t1_pkey PRIMARY KEY (c1)
> );
> CREATE TABLE t2 (
> 	c1 int,
> 	c2 text,
> 	CONSTRAINT t2_pkey PRIMARY KEY (c1)
> );
> 
> CREATE TABLE t4 (
> 	c1 int,
> 	c2 int,
> 	c3 text,
> 	CONSTRAINT t4_pkey PRIMARY KEY (c1)
> );
> 
> INSERT INTO t4
> 	SELECT id,
> 	       id + 1,
> 	       'AAA'
> 	FROM generate_series(1, 100, 1) as t(id);
> DELETE FROM t4 WHERE c1 % 3 != 0;
> 
> INSERT INTO t1
> 	SELECT id,
> 	       id % 10,
> 	       id,
> 	       '1970-01-01'::timestamptz + ((id % 100) || ' days')::interval,
> 	       '1970-01-01'::timestamp + ((id % 100) || ' days')::interval,
> 	       id % 10,
> 	       id % 10,
> 	       1
> 	FROM generate_series(1, 1000, 1) as t(id);
> 
> INSERT INTO t2
> 	SELECT id,
> 	       'AAA'
> 	FROM generate_series(1, 100, 1) t(id);
> 
> SELECT t1.c1, t2.c1 FROM (SELECT c1 FROM t4 WHERE c1 between 50 and 60) t1 FULL JOIN t2 ON (t1.c1 = t2.c1) ORDER BY t1.c1, t2.c1;
> ```

Thanks a lot.

Reply:
> Some suggestion: Maybe it would also be helpful to add a test that evaluates the query and not just looks at the plan. Inspired by the [PSQL tests](https://github.com/postgres/postgres/blob/c8e1ba736b2b9e8c98d37a5b77c4ed31baf94147/contrib/postgres_fdw/expected/postgres_fdw.out):
> 
> ```sql
> CREATE TABLE t1 (
> 	c1 int,
> 	c2 int,
> 	c3 int,
> 	c4 timestamptz,
> 	c5 timestamp,
> 	c6 varchar,
> 	c7 varchar,
> 	c8 int,
> 	CONSTRAINT t1_pkey PRIMARY KEY (c1)
> );
> CREATE TABLE t2 (
> 	c1 int,
> 	c2 text,
> 	CONSTRAINT t2_pkey PRIMARY KEY (c1)
> );
> 
> CREATE TABLE t4 (
> 	c1 int,
> 	c2 int,
> 	c3 text,
> 	CONSTRAINT t4_pkey PRIMARY KEY (c1)
> );
> 
> INSERT INTO t4
> 	SELECT id,
> 	       id + 1,
> 	       'AAA'
> 	FROM generate_series(1, 100, 1) as t(id);
> DELETE FROM t4 WHERE c1 % 3 != 0;
> 
> INSERT INTO t1
> 	SELECT id,
> 	       id % 10,
> 	       id,
> 	       '1970-01-01'::timestamptz + ((id % 100) || ' days')::interval,
> 	       '1970-01-01'::timestamp + ((id % 100) || ' days')::interval,
> 	       id % 10,
> 	       id % 10,
> 	       1
> 	FROM generate_series(1, 1000, 1) as t(id);
> 
> INSERT INTO t2
> 	SELECT id,
> 	       'AAA'
> 	FROM generate_series(1, 100, 1) t(id);
> 
> SELECT t1.c1, t2.c1 FROM (SELECT c1 FROM t4 WHERE c1 between 50 and 60) t1 FULL JOIN t2 ON (t1.c1 = t2.c1) ORDER BY t1.c1, t2.c1;
> ```

Thank you very much.

Reply:
> Some suggestion: Maybe it would also be helpful to add a test that evaluates the query and not just looks at the plan. Inspired by the [PSQL tests](https://github.com/postgres/postgres/blob/c8e1ba736b2b9e8c98d37a5b77c4ed31baf94147/contrib/postgres_fdw/expected/postgres_fdw.out):
> 
> ```sql
> CREATE TABLE t1 (
> 	c1 int,
> 	c2 int,
> 	c3 int,
> 	c4 timestamptz,
> 	c5 timestamp,
> 	c6 varchar,
> 	c7 varchar,
> 	c8 int,
> 	CONSTRAINT t1_pkey PRIMARY KEY (c1)
> );
> CREATE TABLE t2 (
> 	c1 int,
> 	c2 text,
> 	CONSTRAINT t2_pkey PRIMARY KEY (c1)
> );
> 
> CREATE TABLE t4 (
> 	c1 int,
> 	c2 int,
> 	c3 text,
> 	CONSTRAINT t4_pkey PRIMARY KEY (c1)
> );
> 
> INSERT INTO t4
> 	SELECT id,
> 	       id + 1,
> 	       'AAA'
> 	FROM generate_series(1, 100, 1) as t(id);
> DELETE FROM t4 WHERE c1 % 3 != 0;
> 
> INSERT INTO t1
> 	SELECT id,
> 	       id % 10,
> 	       id,
> 	       '1970-01-01'::timestamptz + ((id % 100) || ' days')::interval,
> 	       '1970-01-01'::timestamp + ((id % 100) || ' days')::interval,
> 	       id % 10,
> 	       id % 10,
> 	       1
> 	FROM generate_series(1, 1000, 1) as t(id);
> 
> INSERT INTO t2
> 	SELECT id,
> 	       'AAA'
> 	FROM generate_series(1, 100, 1) t(id);
> 
> SELECT t1.c1, t2.c1 FROM (SELECT c1 FROM t4 WHERE c1 between 50 and 60) t1 FULL JOIN t2 ON (t1.c1 = t2.c1) ORDER BY t1.c1, t2.c1;
> ```

The example is extremely helpful. Thank you very much.